### ISSUE TITLE
Contact tracing on positive test results (trace_on_positive = 1) does not appear to work in new code

### ISSUE THREAD

**Issue by aaron-scheffler** [BDI-pathogens/OpenABM-Covid19#43]
In contexts where contact tracing must be preceded by a positive diagnosis, it is useful to disable tracing on symptom onset (trace_on_symptoms = 0) and instead rely on tracing after a confirmed diagnosis (trace_on_positive = 1). However, the most recent version of code on GitHub cannot do this despite the fact earlier versions (e.g. April 17 2020) were able to accomplish this. Reproducible example below based on code merged from repo today May 7, 2020. The conclusion drawn below is that the app turned on or off does not make an impact when contact tracing is initiated only by a positive diagnosis (total infections remains the same). Given the updates to codes recently, is it possible this functionality was inadvertently disabled?

```
### Scenario 1: App turned off, rely on self-quarantine after seeding infection.
import os
os.getcwd()

path = "/Users/aaron.scheffler/Downloads/Contact_Tracing/OpenABM-Covid19/examples"
os.chdir(path)

import example_utils as utils
params = utils.get_baseline_parameters()
params.set_param( "n_total", 50000 )
params.set_param("trace_on_positive", 1) # Trace on positive test 
params.set_param("trace_on_symptoms", 0) # Do not trace on symptoms (wait for positive test)


sim = utils.get_simulation( params )
sim.steps(1)
while sim.results["total_infected"][ -1] < 500:
    sim.steps(1)
    
sim.env.model.update_running_params("lockdown_on", 1)
sim.steps( 30 )

# lockdown off
sim.env.model.update_running_params("lockdown_on", 0)

# 80% self-quarantine along with their households
sim.env.model.update_running_params("self_quarantine_fraction", 0.8 )
sim.env.model.update_running_params("quarantine_household_on_symptoms", 1 )
  
# turn on the app and quarantine those people who have been traced along with their households
sim.env.model.update_running_params("app_turned_on", 0 ) # app is turned off
sim.env.model.update_running_params("quarantine_on_traced", 1 )
sim.env.model.update_running_params("test_on_symptoms", 1 )

# step forwrard another 100 days
sim.steps( 100 )

import pandas as pd
timeseries = pd.DataFrame( sim.results )
timeseries.plot( x = "time", y = "total_infected" )
timeseries.plot( x = "time", y = "n_tests" )
timeseries.plot( x = "time", y = "n_quarantine" )
```

![image](https://user-images.githubusercontent.com/21348351/81357548-dadba080-9088-11ea-85d1-6721bb518446.png)

![image](https://user-images.githubusercontent.com/21348351/81357565-e16a1800-9088-11ea-9c05-53835bb01408.png)

![image](https://user-images.githubusercontent.com/21348351/81357567-e3cc7200-9088-11ea-8458-53ce893db311.png)

```
### Scenario 2: App turned on, positive tests should initiate tracing and reduce epidemic
import os
os.getcwd()

path = "/Users/aaron.scheffler/Downloads/Contact_Tracing/OpenABM-Covid19/examples"
os.chdir(path)

import example_utils as utils
params = utils.get_baseline_parameters()
params.set_param( "n_total", 50000 )
params.set_param("trace_on_positive", 1) # Trace on positive test 
params.set_param("trace_on_symptoms", 0) # Do not trace on symptoms (wait for positive test)


sim = utils.get_simulation( params )
sim.steps(1)
while sim.results["total_infected"][ -1] < 500:
    sim.steps(1)
    
sim.env.model.update_running_params("lockdown_on", 1)
sim.steps( 30 )

# lockdown off
sim.env.model.update_running_params("lockdown_on", 0)

# 80% self-quarantine along with their households
sim.env.model.update_running_params("self_quarantine_fraction", 0.8 )
sim.env.model.update_running_params("quarantine_household_on_symptoms", 1 )
  
# turn on the app and quarantine those people who have been traced along with their households
sim.env.model.update_running_params("app_turned_on", 1 ) # app is turned on
sim.env.model.update_running_params("quarantine_on_traced", 1 )
sim.env.model.update_running_params("test_on_symptoms", 1 )

# step forwrard another 100 days
sim.steps( 100 )

import pandas as pd
timeseries = pd.DataFrame( sim.results )
timeseries.plot( x = "time", y = "total_infected" )
timeseries.plot( x = "time", y = "n_tests" )
timeseries.plot( x = "time", y = "n_quarantine" )
```
![image](https://user-images.githubusercontent.com/21348351/81357872-b3390800-9089-11ea-8d9f-b7faa2835a20.png)

![image](https://user-images.githubusercontent.com/21348351/81357875-b59b6200-9089-11ea-9571-8acbf8df49b4.png)

![image](https://user-images.githubusercontent.com/21348351/81357882-b8965280-9089-11ea-878c-39e8b53176cd.png)

**Comment by roberthinch** [comment]
Thanks for reporting this, we have noticed it ourselves too and I'll have a PR for the fix tonight hopefully. At the moment if you add the flag `retrace_on_positive = 1` as well it should work as before. (the bug came when we added the ability to convert amber messages to red).

**Comment by aaron-scheffler** [comment]
@roberthinch Wonderful! Your suggestion fixed the problem on my end. Thank you again for your work and responses.

**Comment by roberthinch** [comment]
Fixed in PR #44
